core/module_specifier.rs

```diff
@@ -65,6 +65,18 @@ impl ModuleSpecifier {
     self.0.as_str()
   }
 
+  pub fn to_path(&self) -> PathBuf {
+    if self.0.scheme() == "file" {
+      if let Ok(path) = self.0.to_file_path() {
+        path
+      } else {
+        PathBuf::from(self.0.path())
+      }
+    } else {
+      PathBuf::from(self.0.path())
+    }
+  }
```

Comment:
I'm not happy extending API of `ModuleSpecifier`... It seems this method is only used in single place, so I suggest to just do this logic there.

Reply:
It is only in one place at the moment, but the refactor of the `file_fetcher` also uses it.  It has been one of the biggest frustration I have had of the refactor is that we move from `ModuleSpecifier` to `Url` to `PathBuf` to `Path` to `String`, causing all manner of confusion and indirection.  If we need a `ModuleSpecifier` in a different format, we should do it in one place to ensure it is clear where the logic sits.

Reply:
I will move it for now... 😛

Reply:
Actually looking at it again, it really should be just internal to `MediaType`... which I have made it now.